File: 2015/01/main.ts
#!/usr/bin/env -S deno run --allow-read

const inputText = await Deno.readTextFile(
  new URL("input.txt", import.meta.url),
);

let answerOne = 0;
let answerTwo = 0;

for (const char of inputText) {
  if (char === "(") {
    answerOne++;
  }
  if (char === ")") {
    answerOne--;
  }
}

for (let i = 0; i < inputText.length; i++) {
  if (inputText[i] === "(") {
    answerTwo++;
  }
  if (inputText[i] === ")") {
    answerTwo--;
  }
  if (answerTwo === -1) {
    answerTwo = i + 1;
    break;
  }
}

console.log(`Answer one: ${answerOne}`);
console.log(`Answer two: ${answerTwo}`);

File: 2015/01/main.zig
const std = @import("std");

pub fn main() !void {
    const stdout = std.io.getStdOut().writer();
    const file = try std.fs.cwd().openFile("2015/01/input.txt", .{});
    defer file.close();

    const reader = file.reader();
    var buffer: [1024 * 10]u8 = undefined;

    var answer_one: i32 = 0;
    var answer_two: i128 = 0;
    var answer_two_active: bool = true;

    const read = try reader.read(&buffer);
    for (buffer[0..read], 0..) |char, i| {
        if (char == '(') {
            answer_one += 1;
            if (answer_two_active) answer_two += 1;
        }
        if (char == ')') {
            answer_one -= 1;
            if (answer_two_active) answer_two -= 1;
        }
        if (answer_two_active) {
            if (answer_two == -1) {
                answer_two = i + 1;
                answer_two_active = false;
            }
        }
    }

    try stdout.print("Answer 1: {d}\n", .{answer_one});
    try stdout.print("Answer 2: {d}\n", .{answer_two});
}
